query.cc

```diff
@@ -318,4 +346,19 @@ foreign_ptr<lw_shared_ptr<query::result>> result_merger::get() {
     return make_foreign(make_lw_shared<query::result>(std::move(w), is_short_read, row_count, partition_count));
 }
 
+
+void forward_result::merge(const forward_result& other, const forward_request::reduction_type& type) {
+    switch (type) {
+        case forward_request::reduction_type::count: {
```

Comment:
This won't scale well, especially if we want to support UDAs. I think we should use @avikivity's distributed map-reduce infrastructure instead, and define all built-in aggregates that support map-reduce as a pair of `map` and `reduce` functions. This will be forward compatible with arbitrary user defined map-reduce funtions (via Lua/Wasm).

Reply:
I'm not sure what infrastructure are you talking about. Could you provide some links/pointers to it?

Reply:
I think I wanted to refer to this: https://docs.google.com/document/d/1McmaG--HGtjokIUtL4AnTE-hABMJ71OTOlzx9wmtAvk/edit#
I don't think we ever got beyond an idea phase. In any case, we can merge the code as is and generalize later. Verbs are difficult to change/remove though.

Reply:
The general form is `cql3::expr::expression`. We don't have a way to serialize it now.

I'm okay with adding a new verb if/when we can serialize expressions.